Client/Frontend/Home/ActivityStreamPanel.swift

```diff
@@ -565,7 +570,6 @@ extension ActivityStreamPanel: DataObserverDelegate {
     // Invoked by the ActivityStreamDataObserver when highlights/top sites invalidation is complete.
     func didInvalidateDataSources() {
         self.pendingCacheUpdate = false
-        reloadAll()
```

Comment:
if you do this, anytime you try hiding something from topsites or highlights it wont refresh.

Reply:
I dont think you'll need the `pendingCacheUpdate` bool anymore. It was used to prevent flickering of Topsites if the panel was loading while the cache was being invalidated. I think you can rename it and use it to block the first reloadAll.